js/controller.js

```diff
@@ -1059,7 +1059,7 @@ module.exports = function(OO, _, $, W) {
     },
 
     onPlayed: function() {
-      var duration = this.state.mainVideoDuration;
+      var duration = this.state.mainVideoDuration || this.state.contentTree.duration / 1000;
```

Comment:
Is contentTree always known?
Could be this.state.contentTree.duration === undefined?

Reply:
Also I didn't come up with this line, I borrowed it from onAdsPlayed in which 
`this.state.duration = this.state.contentTree.duration / 1000;` duration gets updated the same way

Reply:
@nzaitcev 
If the response from the backend does not come the value of the duration;
But I thought that in this case most of the logic of the player would be broken.

Reply:
Your second approach also works, but I think this change it's just for cosmetic purposes, I mean, because of the lack of playhead activity after the page reload then the skin will show the zero value on duration field.

Good catch!!